src/GafferImage/Empty.cpp

```diff
@@ -0,0 +1,161 @@
+// Empty implementation
+//////////////////////////////////////////////////////////////////////////
+
+GAFFER_GRAPHCOMPONENT_DEFINE_TYPE( Empty );
+
+size_t Empty::g_firstPlugIndex = 0;
+
+Empty::Empty( const std::string &name )
+	: ImageNode( name )
+{
+	storeIndexOfNextChild( g_firstPlugIndex );
+	addChild( new FormatPlug( "format" ) );
+}
+
+Empty::~Empty()
+{
+}
+
+GafferImage::FormatPlug *Empty::formatPlug()
+{
+	return getChild<FormatPlug>( g_firstPlugIndex );
+}
+
+const GafferImage::FormatPlug *Empty::formatPlug() const
+{
+	return getChild<FormatPlug>( g_firstPlugIndex );
+}
+
+void Empty::affects( const Gaffer::Plug *input, AffectedPlugsContainer &outputs ) const
+{
+	ImageNode::affects( input, outputs );
+
+	if( formatPlug()->displayWindowPlug()->isAncestorOf( input ) )
+	{
+		outputs.push_back( outPlug()->formatPlug() );
+		outputs.push_back( outPlug()->dataWindowPlug() );
+	}
+	else if( input == formatPlug()->pixelAspectPlug() )
+	{
+		outputs.push_back( outPlug()->formatPlug() );
+	}
+}
+
+void Empty::hashFormat( const GafferImage::ImagePlug *output, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	ImageNode::hashFormat( output, context, h );
+	h.append( formatPlug()->hash() );
+}
+
+GafferImage::Format Empty::computeFormat( const Gaffer::Context *context, const ImagePlug *parent ) const
+{
+	return formatPlug()->getValue();
+}
+
+void Empty::hashDataWindow( const GafferImage::ImagePlug *output, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	ImageNode::hashDataWindow( output, context, h );
+	h.append( formatPlug()->hash() );
+}
+
+Imath::Box2i Empty::computeDataWindow( const Gaffer::Context *context, const ImagePlug *parent ) const
+{
+	return formatPlug()->getValue().getDisplayWindow();
+}
+
+IECore::ConstCompoundDataPtr Empty::computeMetadata( const Gaffer::Context *context, const ImagePlug *parent ) const
+{
+	return outPlug()->metadataPlug()->defaultValue();
+}
+
+bool Empty::computeDeep( const Gaffer::Context *context, const ImagePlug *parent ) const
+{
+	return true;
+}
+
+void Empty::hashSampleOffsets( const GafferImage::ImagePlug *output, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	ImagePlug::emptyTileSampleOffsets()->hash( h );
+}
+
+IECore::ConstIntVectorDataPtr Empty::computeSampleOffsets( const Imath::V2i &tileOrigin, const Gaffer::Context *context, const ImagePlug *parent ) const
+{
+	return ImagePlug::emptyTileSampleOffsets();
+}
+
+void Empty::hashChannelNames( const GafferImage::ImagePlug *output, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	ImageNode::hashChannelNames( output, context, h );
+}
+
+IECore::ConstStringVectorDataPtr Empty::computeChannelNames( const Gaffer::Context *context, const ImagePlug *parent ) const
+{
+	IECore::StringVectorDataPtr channelStrVectorData( new IECore::StringVectorData() );
+	std::vector<std::string> &channelStrVector( channelStrVectorData->writable() );
+	channelStrVector.push_back("R");
```

Comment:
Does an empty image need channels at all? `ImagePlug::channelNamesPlug()` has a default value of `[ "R", "G", "B" ]` but I've been meaning to change that to `[]` anyway.

Reply:
Sure, lets define no channels.  The Empty node isn't really useful for anything anyway - as far as I can tell, it's only used in a couple tests.  Seems to still work without any channels.